### PR TITLE
Replace use of properties with features

### PR DESCRIPTION
**Pull request by andy-sweet**
# References and relevant issues
Part of https://github.com/napari/napari/issues/3911

# Description
This replaces use of `properties`, `property_choices`, and `current_properties` with `features` and `feature_defaults`. The motivation for doing this now is to make the latest documentation consistent with existing code and examples on `main` (here and on napari/napari) for the upcoming v0.5 release. As we also plan to deprecate properties in v0.5 (https://github.com/napari/napari/issues/3911), this documentation update is necessary for that.

### CODE REVIEW

**Comment by jni** [comment]
Cool, thanks @andy-sweet! Let me know when you think this is ready for review! 🙏

File: docs/howtos/layers/tracks.md [hunk 1]
```diff
@@ -121,66 +121,6 @@ napari.run()
 * Graph - check this box to display a previously created graph as explained in
   [](#arguments-of-view_tracks-and-add_tracks).
 
-## Arguments of `view_tracks` and `add_tracks`
```
Comment (andy-sweet) [inline docs/howtos/layers/tracks.md:124]: These docs are incomplete and stale for multiple reasons (including properties/features), so I removed this in favor of link to the two functions in the text above.

File: docs/tutorials/annotation/annotate_points.md [hunk 2]
```diff
@@ -4,8 +4,8 @@
 
 ```{Admonition} DEPRECATED ATTRIBUTES
 :class: warning
-As of napari 0.5.0, `edge_*` attributes are being renamed to 
-`border_*` attributes. We have yet to update the images and/or videos in 
+As of napari 0.5.0, `edge_*` attributes are being renamed to
```
Comment (andy-sweet) [inline docs/tutorials/annotation/annotate_points.md:7]: The changes in this file mostly come from the corresponding PR over on the pointannotator repo: https://github.com/kevinyamauchi/PointAnnotator/pull/6

I tested the example on that repo with napari's `main`.

There are also some other name swaps.

File: docs/tutorials/segmentation/annotate_segmentation.md [hunk 3]
```diff
@@ -100,16 +100,16 @@ def circularity(perimeter, area):
 image = data.coins()[50:-50, 50:-50]
 label_image = segment(image)
 
-# create the properties dictionary
-properties = regionprops_table(
+# create the features dictionary
```
Comment (andy-sweet) [inline docs/tutorials/segmentation/annotate_segmentation.md:103]: The code changes come from napari's `examples/annotate_segmentation_with_text.py` example and should be consistent now.

Other changes in this file are mostly word swaps.

File: docs/tutorials/tracking/cell_tracking.md [hunk 4]
```diff
@@ -172,7 +172,7 @@ We can now visualize the full, linked tracks in napari!
 ```python
 viewer = napari.Viewer()
 viewer.add_image(timelapse, scale=SCALE, name='Fluo-N3DH-CE')
-viewer.add_tracks(data, properties=properties, graph=graph, scale=SCALE, name='tracks')
+viewer.add_tracks(data, features=features, graph=graph, scale=SCALE, name='tracks')
```
Comment (andy-sweet) [inline docs/tutorials/tracking/cell_tracking.md:175]: I have not tried upstreaming these changes to https://github.com/quantumjot/btrack/blob/caa56fa82330e4b16b5d28150f9b60ed963165c7/examples/napari_btrack.ipynb

Given their simplicity, I'm fairly confident things should work, but it's probably worth verifying.

**Comment by andy-sweet** [comment]
@jni: I have one last compatibility check to do with the btrack cell-tracking example, but other this is ready for review now.

**Comment by andy-sweet** [comment]
> @jni: I have one last compatibility check to do with the btrack cell-tracking example, but other this is ready for review now.

Check is done.

File: docs/howtos/layers/points.md [hunk 5]
```diff
@@ -435,13 +436,13 @@ nbscreenshot(viewer, alt_text="3 points overlaid on an astronaut image, where th
...
-two properties: `good_point` and `confidence`. The values of each property are
-stored in a NumPy ndarray with length 3 since there were 3 coordinates provided
-in `points`. We set the face color as a function of the `confidence` property by
-providing the keyword argument `face_color='confidence'` to the
-`viewer.add_points()` method. We set the colormap to viridis using the
-`face_colormap` keyword argument as `face_colormap='viridis'`.
+In the example above, the `point_features` were provided as a
+dictionary with two feature columns: `good_point` and `confidence`.
+The table has three rows since there were three coordinates provided in `points`.
+We set the face color as a function of the `confidence` feature by providing the
```
Comment (brisvag) [inline docs/howtos/layers/points.md]: maybe a bit confusing for folks unfamiliar with pandas to talk about columns/rows in a dictionary.

**Review by brisvag — COMMENTED** [review]
Other than the specific comment below, I think there's a general overlap between "table" and "dictionary" which might be confusing to new users.

File: docs/howtos/layers/points.md (hunk 5 above)
Comment (jni) [reply docs/howtos/layers/points.md]: Ah, indeed, maybe worth saying that this is converted internally to a data frame with two columns and three rows? (I wouldn't specify pandas necessarily, but maybe.)

File: docs/tutorials/segmentation/annotate_segmentation.md [hunk 6]
```diff
@@ -355,7 +355,7 @@ text_parameters = {
 }
 ```
 
-The `text` key specifies pattern for the text to be displayed. If `text` is set to the name of a `properties` key, the value for that property will be displayed. napari text also accepts f-string-like syntax, as used here. napari will substitute each pair of curly braces(`{}`) with the values from the property specified inside of the curley braces. For numbers, the precision can be specified in the same style as f-strings. Additionally, napari recognizes standard special characters such as `\n` for new line.
+The `string` key specifies pattern for the text to be displayed. If `string` is set to the name of a `feature` column, the value for that feature will be displayed. napari text also accepts f-string-like syntax, as used here. napari will substitute each pair of curly braces(`{}`) with the values from the feature specified inside of the curly braces. For numbers, the precision can be specified in the same style as f-strings. Additionally, napari recognizes standard special characters such as `\n` for new line.
```
Comment (jni) [inline docs/tutorials/segmentation/annotate_segmentation.md:358]: I always forget about this. It's so cool. 😃

**Review by jni — APPROVED** [review]
Thanks @andy-sweet, this is excellent! I think @brisvag's minor comment is worth addressing, but I'm also ok with it happening later, so I'm gonna approve this and ready-to-merge it, and if you get to it great and if not we can make an issue. 😂

File: docs/howtos/layers/points.md (hunk 5 above)
Comment (andy-sweet) [reply docs/howtos/layers/points.md]: Agreed this paragraph is mixing things a bit too much. Let me try to fix that.

But I also want to make a distinction between the meaning of `features` (i.e. it's a table) and the type used to provide or implement it (i.e. a `dict` or `DataFrame`). Ideally, we could do that in one place for all layers (see https://github.com/napari/docs/issues/23).

For now, there's a little effort above to describe what the table is in the "### Using the points features table" section. I'll clean that up a little and will mention that the table can be provided as a dictionary where the values are array-likes of the same length.

Comment (andy-sweet) [reply docs/howtos/layers/points.md]: I also (tried to) put a link back to the section that describes the dictionary and table/dataframe relationship in both of these similar paragraphs.

**Comment by andy-sweet** [comment]
I think I addressed the feedback and also fixed a couple of links/cross-references. Please take another quick look before merging!

**Review by brisvag — APPROVED** [review]
Love the changes :)

**Comment by andy-sweet** [comment]
Thanks for the second look. I'll merge this after the weekend unless anyone objects.

File: docs/howtos/layers/tracks.md (hunk 1 above)
Comment (jni) [reply docs/howtos/layers/tracks.md:124]: lmao I came here while trying to resolve merge conflicts with #420 going, "why on earth did this get removed. And no discussion either, wth!"

... Then I found my thumbs-up above 🤣

**Pull request merged**